cmonge/trainers/conditional_monge_trainer.py

```diff
@@ -185,7 +198,14 @@ def step_fn(
                 current_logs["eval"] = current_eval_logs
 
             # update state
-            return state_neural_net.apply_gradients(grads=grads), current_logs
+            if (step + 1) % self.grad_acc_steps == 0:
```

Comment:
Step is a traced array, not something that can be converted to boolean. It leads to the following error:
`TracerBoolConversionError: Attempted boolean conversion of traced array with
shape bool[]..
The error occurred while tracing the function step_fn at
/u/adr/Code/conditional-monge/cmonge/trainers/conditional_monge_trainer.py:172
for jit. This concrete value was not available in Python because it depends on
the value of the argument step.
See
https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerBoolConversion
Error`

I will try with state_neural_net.step, since I think this is simply an integer

Reply:
New commit should solve this. Since `step` is a traced array once it is passed to `step_fn, I changed it to the same passing as for `is_logging_step`. So boolean is defined in the `train` function and passes to `step_fn` as a static argument. Passing it as a static argument prevents it from becoming a traced array

Reply:
Great fix, learned about another twist in Jax :) 
Just removed the useless `step` argument fully now